File: BasicsWithJava/ArtBookFragments/app/src/main/java/com/sumere/artbookfragments/view/MainActivity.java
package com.sumere.artbookfragments.view;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.navigation.NavDirections;
import androidx.navigation.Navigation;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;

import com.sumere.artbookfragments.R;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater menuInflater = getMenuInflater();
        menuInflater.inflate(R.menu.menu,menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if(item.getItemId() == R.id.menu_add_art_button){
            com.sumere.artbookfragments.view.ListFragmentDirections.ActionListFragmentToDetailsFragment action = ListFragmentDirections.actionListFragmentToDetailsFragment("new");
            //NavDirections action = ListFragmentDirections.actionListFragmentToDetailsFragment();
            Navigation.findNavController(this,R.id.fragment).navigate(action);
        }
        return super.onOptionsItemSelected(item);
    }
}

File: TestIntroductionKotlin/app/src/main/java/com/sumere/testintroductionkotlin/MainActivity.kt
package com.sumere.testintroductionkotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val tax = Tax()
        var netIncome = tax.calculateIncome(100.0,0.1)
        var netTax = tax.calculateTax(100.0,0.1)

    }
}